### PR TITLE
Feature: Add role page

### LINKED ISSUE #100: Add permission table
Create table for Permissions, and display it in each page (Role, Group and User)

### LINKED ISSUE #109: Create RolePage and display related Users, Groups and Permissions in tabs
Create RolePage and display related Users, Groups and Permissions in tabs

### PR DESCRIPTION
**Pull request by justinehell**
Resolves #100 
Resolves #109 

Check: attach/detach group (or permission) to role → update userStore.permission ??

### CODE REVIEW

File: src/components/tables/PermissionsTable.vue [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+const { t } = useI18n();
+const pagination = ref({
+  rowsPerPage: 0, // infinite
+});
+const displayActionsColumn = computed(
+  () => props.removeAction || props.detachAction,
+);
+const columns = computed(() => {
+  const arrayOfColumns = [{
```
Comment (Zorin95670) [inline src/components/tables/PermissionsTable.vue:88]: Like the key column, add translation for entity and operation.

File: src/i18n/en-US/index.js [hunk 2]
```diff
@@ -165,6 +165,22 @@ export default {
...
+      keyColumn: 'Description',
+      entityColumn: 'Entity',
+      operationColumn: 'Operation',
+      actionsColumn: 'Actions',
+      showAction: 'Show permission',
+      removeAction: 'Remove permission',
+      detachAction: 'Detach permission',
+    },
+    icon: {
+      showAction: 'fa-solid fa-pen-to-square',
```
Comment (Zorin95670) [inline src/i18n/en-US/index.js:200]: to remove

File: src/i18n/en-US/index.js [hunk 3]
```diff
@@ -165,6 +165,22 @@ export default {
...
+      removeAction: 'fa-solid fa-trash',
```
Comment (Zorin95670) [inline src/i18n/en-US/index.js:201]: to remove

File: src/components/tables/PermissionsTable.vue [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          flat
+          rounded
+          color="negative"
+          :icon="t('PermissionsTable.icon.detachAction')"
+          :title="t('PermissionsTable.text.detachAction')"
+          :data-cy="`permission_${cell.row.id}_button_detach`"
+          @click="$emit('detach', cell.row)"
+        />
+        <q-btn
+          v-if="removeAction"
```
Comment (Zorin95670) [inline src/components/tables/PermissionsTable.vue:45]: replace by attach

File: src/services/PermissionService.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Generates permission keys based on the provided array.
+ * @param {Array} permissions - Permissions array.
+ * @param {string} permissions.entity - The entity for which the permission is granted.
+ * @param {string} permissions.action - The action associated with the permission.
+ * @param {string} permissions.libraryId - Optional library ID associated with the permission.
+ * @returns {Array} An array of permission objects with added 'key' property.
+ */
+export function generatePermissionKeys(permissions) {
```
Comment (Zorin95670) [inline src/services/PermissionService.js:11]: generate only one permission key, make this methode return string

File: src/i18n/en-US/index.js (hunk 2 above)
Comment (justinehell) [reply src/i18n/en-US/index.js]: done

File: src/services/PermissionService.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ */
+export async function find() {
+  const api = await prepareRequest();
+
+  return api.get('/permissions').then(({ data }) => {
+    const updatedContent = generatePermissionKeys(data.content);
+
+    return {
+      ...data,
+      content: updatedContent,
```
Comment (Zorin95670) [inline src/services/PermissionService.js:31]:
```js
content: data.content.map((...))
```

File: src/services/PermissionService.js (hunk 5 above)
Comment (justinehell) [reply src/services/PermissionService.js]: done

File: src/services/PermissionService.js (hunk 6 above)
Comment (justinehell) [reply src/services/PermissionService.js]: done

File: src/components/dialog/DetachPermissionFromRoleDialog.vue [hunk 7]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <q-dialog v-model="show">
+    <q-card>
+      <q-card-section class="flex row justify-center">
+        <span class="text-h6 text-center">
+          {{ $t('DetachPermissionFromRoleDialog.text.title',
+                { permission: permission.key, role: role.name }) }}
```
Comment (Zorin95670) [inline src/components/dialog/DetachPermissionFromRoleDialog.vue:7]: { permission: $t{permission.key}, role: role.name }) }}

just an example but give the description of the permission not the key

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**